version: "3.7"
services:
  db:
    image: postgres:9.4
    volumes:
      - pgdata:/var/lib/postgresql/data
    expose:
      - 5432
    environment:
      - POSTGRES_USER=hermes
      - POSTGRES_PASSWORD=hermes
    networks:
      - hermes
    restart: "on-failure"

  db-manager:
    image: hermeshub/db-manager:latest
    expose:
      - 8080
    networks:
      - hermes
    volumes:
      - ./logs/db-manager/:/app/logs
    restart: "on-failure"
    depends_on:
      - db

  event-broker:
    image: redis
    expose:
      - 6379
    networks:
      - hermes
    restart: "always"

  conductor:
    image: hermeshub/conductor:latest
    environment:
      - DOCKER_BINARY_PATH=/usr/local/bin/docker
    ports:
      - 9090:3000
    networks:
      - hermes
    volumes:
      - ./logs/conductor/:/app/logs
      - ./results:/app/results
      - /var/run/docker.sock:/var/run/docker.sock
    restart: "on-failure"
    depends_on:
      - event-broker
      - db-manager

volumes:
  pgdata:

networks:
  hermes:
    external: true
